src/LibraryUtilities.ts

```diff
@@ -245,3 +251,54 @@ export function buildLibraryItemsFromLayoutSpecs(loadedTypes: any, layoutSpecs:
     }
     return convertToLibraryTree(typeListNodes, layoutElements);
 }
+
+// Recursively set visible and expanded of ItemData back to default 
+export function resetItemData(items: ItemData[]) {
+    for(let item of items) {
+        item.visible = true;
+        item.expanded = false;
+        resetItemData(item.childItems);
+    }
+}
+
+export function showItemRecursive(item: ItemData) {
+    item.visible = true;
+    item.expanded = true;
+    for(let childItem of item.childItems) {
+        showItemRecursive(childItem);
+    }
+}
```

Comment:
Hi @fanwgwg, thanks for doing this. In a subsequent pull request you can merge `resetItemData` and `showItemRecursive` into one that looks like this:

```js
export function setItemStateRecursive(items: ItemData[], visible: boolean, expanded: boolean) {
    // Makes use of visible and expanded parameters
}
```